File: ruby/easy/construct_the_rectangle.rb
# @param {Integer} area
# @return {Integer[]}
def construct_rectangle(area)
    sqrt, min, min_pair = Math.sqrt(area), area, [1,area]
    for i in 1..sqrt
        if (area % i) == 0
            div = (area/i)
            if (i - div).abs < min
                min = (i - div).abs
                min_pair = [[i, div].max, [i,div].min]
            end
        end
    end
    return min_pair
end

File: javascript/easy/constructtheRectangle.js
/**
 * @param {number} area
 * @return {number[]}
 */
var constructRectangle = function(area) {
    const sqrt = parseInt(Math.sqrt(area));
    let min = area, minPair = [area, 1];
    for(let i = 1; i <= sqrt; i++) {
        if((area % i) === 0) {
            let div = parseInt(area/i), diff = Math.abs(div);
            if(diff < min) {
                min = diff;
                minPair = (i > div) ? [i,div] : [div,i];
            }
        }
    }
    return minPair;
};
